tests/cases/compiler/isolatedModulesNoExternalModuleMultiple.ts

```diff
@@ -0,0 +1,11 @@
+// @isolatedModules: true
+// @target: es6

```

Comment:
Could you also add a test with `moduleDetection: force`? Optionally you could also try to add a test `moduleDetection: auto` and with `package.json` containing `{"type":"module"}`. Note that I'm not super familiar with this new option and the latter proposed test might also require `moduleResolution: node16` (but I'm not sure about it)

Reply:
Added a test case with `isolatedModules` + `moduleDetection: force`. I don't expect a regression in this area from the primary fix here, but it seems like a good test to have.

Did not add other test cases beyond that; I am not too familiar with the newer Node-related resolution behaviors, and I don't expect any regression around those.